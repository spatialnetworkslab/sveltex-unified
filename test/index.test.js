import dedent from 'dedent'
import { processor } from '../src/index.js'

const process = md => {
  return processor()
    .processSync(md)
    .toString()
}

test('process basic markdown correctly', () => {
  const input = process(dedent`
      # test
  `)
  const output = dedent`
      <h1>test</h1>
  `
  expect(input).toBe(output)
})

test('process math inline equation', () => {
  const input = process(dedent`
      Inline equation $x + 1$
  `)
  expect(input).toMatchSnapshot()
})

test('process math block equation', () => {
  const input = process(dedent`
      Using a block equation:

      $$
      x + 1
      $$

      Continue with normal paragraph
  `)
  expect(input).toMatchSnapshot()
})

test('highlight code block syntax', () => {
  const input = process(dedent`
  \`\`\`js
  var x = 1
  \`\`\`
  `)
  expect(input).toMatchSnapshot()
})

test('highlight code block html syntax', () => {
  const input = process(dedent`
  \`\`\`html
  <li>my list item</li>
  \`\`\`
  `)
  expect(input).toMatchSnapshot()
})

test('highlight svelte syntax', () => {
  const input = process(dedent`
  \`\`\`svelte
    <script>
      let x = 7;
    </script>

    {#if x > 10}
      <p>{x} is greater than 10</p>
    {:else if 5 > x}
      <p>{x} is less than 5</p>
    {:else}
      <p>{x} is between 5 and 10</p>
    {/if}
  \`\`\`
  `)
  expect(input).toMatchSnapshot()
})

test('curlies in code blocks are escaped', () => {
  const input = process(dedent`
      \`\`\`svelte
      {#if x > 10}
        <p>{x} is greater than 10</p>
      {/if}
    \`\`\`
  `)
  expect(input).toMatchSnapshot()
})

test('paragraphs within else-if blocks should be terminated', () => {
  const input = process(dedent`
    {#if a < 10}
    One
    {:else if a == 10}
    Two
    {:else}
    Three
    {/if}
  `)
  expect(input).toMatchSnapshot()
})
