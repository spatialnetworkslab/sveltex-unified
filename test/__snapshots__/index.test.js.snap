// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`highlight code block html syntax 1`] = `
"<pre class=\\"language-html\\"><code class=\\"language-html\\"><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;</span>li</span><span class=\\"token punctuation\\">></span></span>my list item<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;/</span>li</span><span class=\\"token punctuation\\">></span></span>
</code></pre>"
`;

exports[`highlight code block syntax 1`] = `
"<pre class=\\"language-js\\"><code class=\\"language-js\\"><span class=\\"token keyword\\">var</span> x <span class=\\"token operator\\">=</span> <span class=\\"token number\\">1</span>
</code></pre>"
`;

exports[`highlight svelte syntax 1`] = `
"<pre class=\\"language-svelte\\"><code class=\\"language-svelte\\">  <span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;</span>script</span><span class=\\"token punctuation\\">></span></span><span class=\\"token script\\"><span class=\\"token language-javascript\\">
    <span class=\\"token keyword\\">let</span> x <span class=\\"token operator\\">=</span> <span class=\\"token number\\">7</span><span class=\\"token punctuation\\">;</span>
  </span></span><span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;/</span>script</span><span class=\\"token punctuation\\">></span></span>

  <span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span>#<span class=\\"token keyword\\">if</span> x <span class=\\"token operator\\">></span> <span class=\\"token number\\">10</span><span class=\\"token punctuation\\">}</span></span>
    <span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;</span>p</span><span class=\\"token punctuation\\">></span></span><span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span>x<span class=\\"token punctuation\\">}</span></span> is greater than 10<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;/</span>p</span><span class=\\"token punctuation\\">></span></span>
  <span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span><span class=\\"token punctuation\\">:</span><span class=\\"token keyword\\">else</span> <span class=\\"token keyword\\">if</span> <span class=\\"token number\\">5</span> <span class=\\"token operator\\">></span> x<span class=\\"token punctuation\\">}</span></span>
    <span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;</span>p</span><span class=\\"token punctuation\\">></span></span><span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span>x<span class=\\"token punctuation\\">}</span></span> is less than 5<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;/</span>p</span><span class=\\"token punctuation\\">></span></span>
  <span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span><span class=\\"token punctuation\\">:</span><span class=\\"token keyword\\">else</span><span class=\\"token punctuation\\">}</span></span>
    <span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;</span>p</span><span class=\\"token punctuation\\">></span></span><span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span>x<span class=\\"token punctuation\\">}</span></span> is between 5 and 10<span class=\\"token tag\\"><span class=\\"token tag\\"><span class=\\"token punctuation\\">&#x3C;/</span>p</span><span class=\\"token punctuation\\">></span></span>
  <span class=\\"token language-javascript\\"><span class=\\"token punctuation\\">{</span><span class=\\"token operator\\">/</span><span class=\\"token keyword\\">if</span><span class=\\"token punctuation\\">}</span></span>
</code></pre>"
`;

exports[`paragraphs within else-if blocks should be terminated 1`] = `
"<p>{#if a &#x3C; 10}
One
{:else if a == 10}
Two
{:else}
Three
{/if}</p>"
`;

exports[`process math block equation 1`] = `
"<p>Using a block equation:</p>
<div class=\\"math math-display\\"><span class=\\"katex-display\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>x</mi><mo>+</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">x + 1</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.66666em;vertical-align:-0.08333em;\\"></span><span class=\\"mord mathdefault\\">x</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span></span></span></span></span></div>
<p>Continue with normal paragraph</p>"
`;

exports[`process math inline equation 1`] = `"<p>Inline equation <span class=\\"math math-inline\\"><span class=\\"katex\\"><span class=\\"katex-mathml\\"><math xmlns=\\"http://www.w3.org/1998/Math/MathML\\"><semantics><mrow><mi>x</mi><mo>+</mo><mn>1</mn></mrow><annotation encoding=\\"application/x-tex\\">x + 1</annotation></semantics></math></span><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.66666em;vertical-align:-0.08333em;\\"></span><span class=\\"mord mathdefault\\">x</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span><span class=\\"mbin\\">+</span><span class=\\"mspace\\" style=\\"margin-right:0.2222222222222222em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.64444em;vertical-align:0em;\\"></span><span class=\\"mord\\">1</span></span></span></span></span></p>"`;
